src/marks/auto.js

```diff
@@ -211,23 +164,78 @@ export function auto(data, options) {
   if (transform) {
     if (transform === bin || transform === binX) markOptions.x = {value: X, ...xOptions};
     if (transform === bin || transform === binY) markOptions.y = {value: Y, ...yOptions};
-    markOptions = transform(transformOptions, markOptions);
   }
 
   // If zero-ness is not specified, default based on whether the resolved mark
-  // type will include a zero baseline. TODO Move this to autoSpec.
+  // type will include a zero baseline.
   if (xZero === undefined)
-    xZero = X && transform !== binX && (mark === barX || mark === areaX || mark === rectX || mark === ruleY);
+    xZero =
+      X &&
+      transform !== bin &&
+      transform !== binX &&
+      (markImpl === barX || markImpl === areaX || markImpl === rectX || markImpl === ruleY);
   if (yZero === undefined)
-    yZero = Y && transform !== binY && (mark === barY || mark === areaY || mark === rectY || mark === ruleX);
+    yZero =
+      Y &&
+      transform !== bin &&
+      transform !== binY &&
+      (markImpl === barY || markImpl === areaY || markImpl === rectY || markImpl === ruleX);
+
+  return {
+    fx: fx ?? null,
+    fy: fy ?? null,
+    x: {
+      value: xValue ?? null,
+      reduce: xReduce ?? null,
+      zero: xZero ?? false,
```

Comment:
Might as well coerce the input to a boolean here, in case the user passed in something like `0`?

```suggestion
      zero: !!xZero,
```